db/db_impl/db_impl_write.cc

```diff
@@ -1897,19 +1897,13 @@ Status DBImpl::DelayWrite(uint64_t num_bytes, WriteThread& write_thread,
 
 // REQUIRES: mutex_ is held
 // REQUIRES: this thread is currently at the front of the writer queue
-void DBImpl::WriteBufferManagerStallWrites() {
+void DBImpl::MaybeWriteBufferManagerStallWrites() {
   mutex_.AssertHeld();
   // First block future writer threads who want to add themselves to the queue
   // of WriteThread.
   write_thread_.BeginWriteStall();
   mutex_.Unlock();
-
-  // Change the state to State::Blocked.
-  static_cast<WBMStallInterface*>(wbm_stall_.get())
-      ->SetState(WBMStallInterface::State::BLOCKED);
```

Comment:
Cleanup 8: make this part of the responsibility of MaybeBeginWriteStall() as the casting `static_cast<WBMStallInterface*>` indicates we are dealing something too low-level here